packages/flutter/lib/src/widgets/interactive_viewer.dart

```diff
@@ -924,6 +924,7 @@ class _InteractiveViewerState extends State<InteractiveViewer> with TickerProvid
       }
       if (widget.onInteractionUpdate != null) {
         widget.onInteractionUpdate!(ScaleUpdateDetails(
+          focalPoint: _transformationController!.toScene(event.localPosition),
```

Comment:
According to the documentation

> The coordinates returned in the details are viewport coordinates relative to the parent.

Is that correct?

Reply:
No it's not, thanks for catching that!  I was just copying what I did for the other onInteractionUpdate, which is also wrong apparently.

I will change the code in both places so that the documentation is correct. I think that's best because the user can still convert to scene coordinates themselves using the same toScene method.

Reply:
Done.  I also updated a test to make sure this is covered.

Reply:
`onInteractionStart` should also use the viewport coordinate system according to the documentation? (`onInteractionStart` is called after updating the transform. Is that expected?) 

also nit: I think we can do `widget.onInteractionUpdate?.call(ScaleUpdateDetails(...))` instead?

Reply:
Oh yeah that too.  I've now made sure that all of the onInteraction* calls are using the coordinates of the event (not transformed to the scene) and that they are called after the controller has been updated.  That way the user can transform them to scene coordinates if they want by using the controller, but otherwise have the viewport coordinates.  That seems like the best approach to me.   I also wrote another test confirming that you can do this.

Reply:
Someone actually opened an issue for this (https://github.com/flutter/flutter/issues/63606). This PR should close that issue too now.